docs/features/techdocs/using-cloud-storage.md

```diff
@@ -309,3 +310,77 @@ and read the static generated documentation files. When you start the backend of
 the app, you should be able to see
 `techdocs info Successfully connected to the Azure Blob Storage container` in
 the logs.
+
+## Configuring OpenStack Swift Container with TechDocs
+
+Follow the
+[official OpenStack Api documentation](https://docs.openstack.org/api-ref/identity/v3/)
+for the latest instructions on the following steps involving Open Stack Storage.
+
+**1. Set `techdocs.publisher.type` config in your `app-config.yaml`**
+
+Set `techdocs.publisher.type` to `'openStackSwift'`.
+
+```yaml
+techdocs:
+  publisher:
+    type: 'openStackSwift'
+```
+
+**2. Create an OpenStack Swift Storage Container**
+
+Create a dedicated container for TechDocs sites.
+[Refer to the official documentation](https://docs.openstack.org/mitaka/user-guide/dashboard_manage_containers.html).
+
+TechDocs will publish documentation to this container and will fetch files from
+here to serve documentation in Backstage. Note that the container names are
+globally unique.
+
+Set the config `techdocs.publisher.openStackSwift.containerName` in your
+`app-config.yaml` to the name of the container you just created.
+
+```yaml
+techdocs:
+  publisher:
+    type: 'openStackSwift'
+    openStackSwift:
+      containerName: 'name-of-techdocs-storage-container'
+```
+
+**3. Authentication using app-config.yaml**
+
+Set the configs in your `app-config.yaml` to the your container name.
+
+https://docs.openstack.org/api-ref/identity/v3/?expanded=password-authentication-with-unscoped-authorization-detail#password-authentication-with-unscoped-authorization
+for more details.
+
+```yaml
+techdocs:
+  publisher:
+    type: 'openStackSwift'
+    openStackSwift:
+      containerName: 'name-of-techdocs-storage-bucket'
+      credentials:
+        userName:
+          $env: OPENSTACK_SWIFT_STORAGE_USERNAME
+        password:
+          $env: OPENSTACK_SWIFT_STORAGE_PASSWORD
+      authUrl:
+        $env: OPENSTACK_SWIFT_STORAGE_AUTH_URL
+      keystoneAuthVersion:
+        $env: OPENSTACK_SWIFT_STORAGE_AUTH_VERSION
+      domainId:
+        $env: OPENSTACK_SWIFT_STORAGE_DOMAIN_ID
+      domainName:
+        $env: OPENSTACK_SWIFT_STORAGE_DOMAIN_NAME
+      region: 
+        $env: OPENSTACK_SWIFT_STORAGE_REGION
```

Comment:
As we discussed on the group call, it's fine to start with adding these credentials in the config, but ideally they should follow the pattern of AWS/GCS publisher where we don't define anything in the config, and the client reads them from environment variables.

Reply:
Is that because of the usage in the CLI? The backend also needs to be authenticated to read the data back though right? And we'd definitely want to use the config for that. Prolly makes sense to use this `fromConfig` in the backend but the constructor in the CLI in that case?

Reply:
> And we'd definitely want to use the config for that.

Not always. In case of S3 for example, if there is no credentials defined in the config, we create an instance of `aws-sdk` S3 client with empty parameters. The client is responsible for reading the environment variables on its own.

In case of OpenStack Swift, I am not sure how well the `pkgcloud` can read environment variables. Hence, I thought it's fine to explicitly define all these configs, but later on investigate to remove as much as possible, and leverage the environment variables that the relevant SDK client expects.


> Prolly makes sense to use this fromConfig in the backend but the constructor in the CLI in that case?

Yeah could be done.